pkg/storagepool/cstorpool.go

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright 2017 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storagepool
+
+import (
+	"fmt"
+
+	"github.com/ghodss/yaml"
+	"github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	m_k8s_client "github.com/openebs/maya/pkg/client/k8s"
+	mach_apis_meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"github.com/openebs/maya/pkg/engine"
+)
+
+// cstorPool OperationOptions contains the options with respect to
+// cstorPool related operations
+type cstorPoolOperationOptions struct {
+	// runNamespace is the namespace where cstorPool operation will happen
+	//runNamespace string
+	// k8sClient will make K8s API calls
+	k8sClient *m_k8s_client.K8sClient
+}
+
+// cstorPoolOperation exposes methods with respect to cstorPool related operations
+// e.g. read, create, delete.
+type cstorPoolOperation struct {
+	// cstorPoolOperationOptions has the options to various cstorPool related
+	// operations
+	cstorPoolOperationOptions
+	// cstorPool to create or read or delete
+	cstorPool *v1alpha1.CStorPool
+}
+
+// NewCstorPoolOperation returns a new instance of cstorPoolOperation
+func NewCstorPoolOperation(cstorPool *v1alpha1.CStorPool) (*cstorPoolOperation, error) {
+	if cstorPool == nil {
+		return nil, fmt.Errorf("failed to instantiate cstorPool operation: nil cstorPool was provided")
+	}
+
+	kc, err := m_k8s_client.NewK8sClient(cstorPool.Namespace)
+	if err != nil {
+		return nil, err
+	}
+	// Put cstor pool object inside cstorPoolOperation object
+	return &cstorPoolOperation{
+		cstorPool: cstorPool,
+		cstorPoolOperationOptions: cstorPoolOperationOptions{
+			k8sClient: kc,
+		},
+	}, nil
+}
+
+// Create provisions an OpenEBS cstorPool
+func (v *cstorPoolOperation) Create() (*v1alpha1.CStorPool, error) {
+	if v.k8sClient == nil {
+		return nil, fmt.Errorf("unable to create cstorPool: nil k8s client")
+	}
+
+	castName := v.cstorPool.Annotations[string(v1alpha1.SPCASTemplateCK)]
+	if len(castName) == 0 {
+		//return nil, fmt.Errorf("unable to create cstorPool: missing create cas template at '%s'", v1alpha1.CASTemplateCVK)
+		return nil, fmt.Errorf("unable to create cstorPool: missing create cas template")
```

Comment:
reword the message:
"failed to create cstorpool: missing cas template to create cstorpool"